ibis/expr/rules.py

```diff
@@ -794,3 +799,125 @@ def _validate(self, args, i):
 
 def comparable(left, right):
     return ir.castable(left, right) or ir.castable(right, left)
+
+
+class Table(Argument):
+    """A table argument.
+
+    Parameters
+    ----------
+    name : str
+        The name of the table argument.
+    optional : bool
+        Whether this table argument is optional or not.
+    satisfying : iterable
+        An iterable of lambda expressions, which will be used to validate
+        arguments. Each lambda expression must take a table as its single
+        argument, validate, and then return ``True`` for tables that
+        pass validation and ``False`` otherwise.
```

Comment:
is it possible to have satisfying or schema (or both as you have them now)?

iow what if i don’t care about a specific schema
i could validate in satisfying (or do some general rule inside that)